src/look_ahead.rs

```diff
@@ -16,63 +16,68 @@ impl<'a> Lookahead<'a> {
     ) -> Self {
         Self {
             fragments,
-            field: Some(field),
+            fields: vec![field],
         }
     }
 
-    /// Get the first subfield of the selection set with the specified name. This will ignore
+    /// Get the field of the selection set with the specified name. This will ignore
     /// aliases.
     ///
     /// For example, calling `.field("a")` on `{ a { b } }` will return a lookahead that
     /// represents `{ b }`.
     pub fn field(&self, name: &str) -> Self {
+        let mut fields = Vec::new();
+        for field in &self.fields {
+            filter(&mut fields, self.fragments, &field.selection_set.node, name)
+        }
+
         Self {
             fragments: self.fragments,
-            field: self
-                .field
-                .and_then(|field| find(self.fragments, &field.selection_set.node, name)),
+            fields,
         }
     }
 
     /// Returns true if field exists otherwise return false.
     #[inline]
     pub fn exists(&self) -> bool {
-        self.field.is_some()
+        !self.fields.is_empty()
     }
 }
 
 impl<'a> From<SelectionField<'a>> for Lookahead<'a> {
     fn from(selection_field: SelectionField<'a>) -> Self {
         Lookahead {
             fragments: selection_field.fragments,
-            field: Some(selection_field.field),
+            fields: vec![selection_field.field],
         }
     }
 }
 
-fn find<'a>(
+fn filter<'a>(
+    fields: &mut Vec<&'a Field>,
     fragments: &'a HashMap<Name, Positioned<FragmentDefinition>>,
     selection_set: &'a SelectionSet,
     name: &str,
-) -> Option<&'a Field> {
-    selection_set
-        .items
-        .iter()
-        .find_map(|item| match &item.node {
+) {
+    for item in &selection_set.items {
```

Comment:
I've done this imperatively. Tried doing this with iterators but I'm pretty sure you need to use a `Box<dyn Iterator>` because it's recursive. Happy to change if there's a better way though.